content/doc/tutorials/building-a-python-app-with-pyinstaller.adoc

```diff
@@ -0,0 +1,486 @@
+application with PyInstaller in Jenkins. Your Pipeline will be created as a
+`Jenkinsfile`, which will be committed to your locally cloned Git repository
+(`simple-python-pyinstaller-app`).
+
+This is the foundation of "Pipeline-as-Code", which treats the continuous
+delivery pipeline a part of the application to be versioned and reviewed like
+any other code. Read more about Pipeline and what a Jenkinsfile is in the
+link:/doc/book/pipeline[Pipeline] and link:/doc/book/pipeline/jenkinsfile[Using
+a Jenkinsfile] sections of the User Handbook.
+
+First, create an initial Pipeline with a "Build" stage that executes the first
+part of the entire production process for your application. This "Build" stage
+downloads a Python Docker image and runs it as a Docker container, which in turn
+compiles your simple Python application into byte code.
+
+. Using your favorite text editor or IDE, create and save new text file with the
+  name `Jenkinsfile` at the root of your local `simple-python-pyinstaller-app`
+  Git repository.
+. Copy the following Declarative Pipeline code and paste it into your empty
+  `Jenkinsfile`:
++
+[source,groovy]
+----
+pipeline {
+    agent none // <1>
+    stages {
+        stage('Build') { // <2>
+            agent {
+                docker {
+                    image 'python:3.4-slim' // <3>
+                }
+            }
+            steps {
+                sh 'python -m py_compile sources/add2vals.py' // <4>
+            }
+        }
+    }
+}
+----
+<1> The link:/doc/book/pipeline/syntax#agent[`agent`] section with the `none`
+parameter specified at the top of this Pipeline code block means that no global
+agent will be allocated for the entire Pipeline's execution and that each
+link:/doc/book/pipeline/syntax/#stage[`stage`] directive must specify its own
+`agent` section.
+<2> Defines a link:/doc/book/pipeline/syntax/#stage[`stage`] (directive) called
+`Build` that appears on the Jenkins UI.
+<3> This `image` parameter (of the link:/doc/book/pipeline/syntax#agent[`agent`]
+section's `docker` parameter) downloads the
+https://hub.docker.com/_/python/[`python:3.4-slim` Docker image] (if it's not
+already available in your Docker host) and runs this image as a separate
+container. This means that:
+* You'll have separate Jenkins/Blue Ocean and Python containers running locally
+  in Docker.
+* The Python container becomes the link:/doc/book/glossary/#agent[agent] that
+  Jenkins uses to run the `Build` stage of your Pipeline project. However,
+  this container is short-lived - its lifespan is only that of the duration of
+  your `Build` stage's execution.
+<4> This `sh` step (of the link:/doc/book/pipeline/syntax/#steps[`steps`]
+section) runs the Python command to compile your application into a byte code
+file (with a `.pyc` extension), which is placed into the `sources/__pycache__`
+workspace directory (i.e. within the
+`/var/jenkins_home/workspace/simple-python-pyinstaller-app` directory in the
```

Comment:
This "i.e." clause seems like an unnecessary implementation detail for the reader, I suggest removing it

Reply:
Done (across all current/relevant tutorials)